src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,47 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index > -1) {
+      Animal.alive.splice(index, 1);
+    }
```

Comment:
The use of `indexOf` and `splice` inside the `die` method could lead to performance issues if there are many animals in the `Animal.alive` array. Consider using `filter` to create a new array without the dead animal, which avoids nested loops and is generally more performant.